Python/Product/Analysis/ModulePath.cs

```diff
@@ -664,17 +659,21 @@ out errorParameter
         internal static bool FromBasePathAndName_NoThrow(
             string basePath,
             string moduleName,
+            PythonLanguageVersion? languageVersion,
             out ModulePath modulePath
         ) {
-            return FromBasePathAndName_NoThrow(basePath, moduleName, null, null, out modulePath, out _, out _, out _);
+            var isPackage = languageVersion.HasValue && PythonVersionRequiresInitPyFiles(languageVersion.Value) ? (Func<string, bool>)null : s => true;
```

Comment:
For least surprise, we should assume that `__init__.py` files are required if we don't know the language version. (That's the recommended way to create packages, and it works in all versions, whereas implicit namespace packages only work in recent versions and are only recommended for certain cases.)